src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php

```diff
@@ -15,17 +15,18 @@
 use Illuminate\Support\Str;
 use InvalidArgumentException;
 
+/** @mixin \Illuminate\Database\Eloquent\Builder */
 trait QueriesRelationships
 {
     /**
      * Add a relationship count / exists condition to the query.
      *
-     * @param  \Illuminate\Database\Eloquent\Relations\Relation|string  $relation
+     * @param  \Illuminate\Database\Eloquent\Relations\Relation<*, *, *>|string  $relation
```

Comment:
After adding this doc-block in `has`, `whereHas` and other similar functions those started showing warning for string 👀

Something like... It accepts `Relation but string given` 🤔

It's only me?

Reply:
There's a passing test for this case:

https://github.com/laravel/framework/blob/6c16f47db630b03837683af9f9fb3a1f28d9e0dd/types/Database/Eloquent/Builder.php#L63

What does your code look like? And if you're using Larastan then could be an issue on that side of things

Reply:
Something like this 
```php
Model::query()
   ->whereHas('name', function(Builder $query){
$query->where();
}
```

Reply:
And what is the full error message, I'm unable to replicate in the framework type tests...

Reply:
not breaking but the message is `Argument '1' passed to whereHas() is expected to be of type Illuminate\Database\Eloquent\Relations\Relation, string givenPHP(PHP0406)`

Reply:
Given that the tests are passing and the methods do have the proper types, this would appear to be an issue with your environment. The most likely cause is that there's a stale result cache, or a stub that's overriding the method (ide-helper?).

Reply:
Okay, fyi I changed 3 laptops and 5 projects with the same results.
Windows and VsCode.

Note: everything is working fine. Phpstan 9 passing.

It's an IDE issue but it comes with an update.